clients/cobol-lsp-vscode-extension/src/services/reporter/TelemetryService.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import {userInfo} from "os";
+import {sep} from "path";
+import {TelemetryEvent} from "./model/TelemetryEvent";
+import {TelemetryFactory} from "./TelemetryFactory";
+
+export class TelemetryService {
+    /**
+     *  This method calculate the time elapsed between two number representation of dates expressed as
+     *  number of milliseconds elapsed since January 1, 1970 00:00:00 UTC.
+     * @param startTime numeric representation of a starting time
+     * @param endTime numeric representation of a ending time
+     */
+    public static calculateTimeElapsed(startTime: number, endTime: number): number | undefined {
+        return ((startTime && endTime) && (endTime >= startTime)) ? ((endTime - startTime) / 1000) : undefined;
+    }
+
+    /**
+     * This method collect the data event for telemetry purpose, apply basic validation
+     * and delegate the reporter service to sent them to the target telemetry server
+     * @param eventName name assigned for the event for the telemetry
+     * @param categories optional list of keywords (act as tags)
+     * @param notes optional brief description
+     * @param telemetryMeasurement optional set of numeric data with a key name
+     */
+    public static registerEvent(eventName: string, categories?: string[], notes?: string, telemetryMeasurement?: Map<string, number>) {
+        if (!this.isValidEventName(eventName)) {
+            return;
+        }
+        TelemetryFactory.getReporter().reportEvent(this.createTelemetryEvent(eventName, categories, notes, undefined, telemetryMeasurement));
+
+    }
+
+    /**
+     * This method collect the exception data event for telemetry purpose, apply basic validation
+     * and delegate the reporter service to sent them to the target telemetry server
+     * @param eventName name assigned for the event for the telemetry
+     * @param rootCause the stacktrace attached for debugging purpose.
+     * @param categories optional list of keywords (act as tags)
+     * @param notes optional brief description
+     * @param telemetryMeasurement optional set of numeric data with a key name
+     */
+    public static registerExceptionEvent(eventName = "RuntimeException", rootCause: string, categories?: string[], notes?: string, telemetryMeasurement?: Map<string, number>) {
+        if (!(this.isValidEventName(eventName) && this.isValidRootCause(rootCause))) {
+            return;
```

Comment:
shortened faster-form: 
if (this.isValidEventName(eventName) && this.isValidRootCause(rootCause)) {
             TelemetryFactory.getReporter(). .. 
        }